android_17/src/product_list.js

```diff
@@ -1,28 +1,62 @@
 import React from 'react';
-import ProductData from './product_data';
-import PRODUCT_DATA from './product_data';
+
 
 export class ProductList extends React.Component {
 
+    constructor() {
+        super();
+        this.state = {
+            products: [],
+            isLoaded: false,
+        };
+    }
+
+    componentDidMount() {
+        fetch('http://localhost:5000/v1/products')
+            .then(res => res.json())
+            .then(json => {
+                this.setState({
+                    isLoaded: true,
+                    products: json['response'],
+                })
+            });
+    }
+
+
 
     render() {
+       console.log(this.state.products)
 
+        var {  isLoaded, products } = this.state;
 
+        if(!isLoaded) {
+            return <div>Loading...</div>;
+        }
+
+        else {
+        
         return (
             <div>
                 <ul>
-                    {PRODUCT_DATA.map(item => <li key={item.id}>
-                        {item.name}
-                    </li>)}
-                    {PRODUCT_DATA.map(item => <li key={item.id}>
-                        {item.price}
-                    </li>)}
+                    {products.map(product => (
+                        <li key={product.sid}>
+                            <div>
+                                <p>{product.name}</p>
+                            </div>
+                            <div>{product.price}</div>
+                        </li>
+                    ))}
                 </ul>
+
             </div>
-        )
+        );
     }
 }
 
+
```

Comment:
Can you get rid of these two blank lines here? Just want to clean thngs up a bit